components/PlaybackTask.brs

```diff
@@ -0,0 +1,244 @@
+
+  raf.SetDebugOutput(false) 'debugging
+  raf.SetAdPrefs(false)
+
+  ' adUrl = "pkg:/res/vast.xml"  ' Regular Ads
+  adUrl = "pkg:/res/vast-truex.xml"  ' TrueX preroll + Ads
+  raf.setAdUrl(adUrl)
+  
+  m.adPods = raf.getAds()
+
+  m.raf = raf
+end sub
+
+sub startPlayback()
+  ? "TRUE[X] >>> startPlayback()"
+
+  checkPreroll() ' Currently doesn't handle secondary ad flows (eg. skip choice card).  Requires letting playback to handle
+  if NOT m.isInTruexAd
+    m.videoPlayer.control = "play" 'TODO: Should this go in the flow?
+  end if
+
+  while(true)
+    msg = Wait(0, m.port)
+    msgType = type(msg)
+
+    ' ? "event type:" msgType
+    if msgType = "roSGNodeEvent"
+      field = msg.getField()
+      ? "roSGNodeEvent msg.getField()" field
+      if field = "position" then 
+        onPositionChanged(msg)
+      else if field = "event" then
+        onTrueXEvent(msg)
+      end if
+      ' TODO: Video complete event
+    end if
+  end while
+end sub
+
+sub checkPreroll()
+  ads = m.raf.getAds()
+  if ads = invalid then return
+
+  for each adPod in ads
+    if adPod.rendersequence = "preroll"
+      handleAds(adPod)
+      exit for
+    end if
+  end for
+end sub
+
+sub onPositionChanged(event)
+  position = event.getData()
+  ' ? "onPositionChanged: " position
+
+  ads = m.raf.getAds(event)
+  handleAds(ads)
+end sub
+
+sub onTruexEvent(event)
+  data = event.getData()
+  eventType = data.type
+
+  ? "TRUE[X] >>> PlaybackTask::onTrueXEvent(): " eventType
+
+  types = {
+    "ADSTARTED": "adStarted",
+    "ADCOMPLETED": "adCompleted"
+    "ADFIRSTQUARTILE": "adFirstQuartile",
+    "ADSECONDQUARTILE": "adSecondQuartile",
+    "ADTHIRDQUARTILE": "adThirdQuartile",
+    "ADPOSITION": "adPosition",
+    "ADERROR": "adError",
+    "ADFREEPOD": "adFreePod"
+    "NOADSAVAILABLE": "noAdsAvailable",
+    "OPTIN": "optIn",
+    "OPTOUT": "optOut",
+    "USERCANCEL": "userCancel",
+    "USERCANCELSTREAM": "userCancelStream",
+    "SKIPCARDSHOWN": "skipCardShown"
+  }
+
+  ' TODO: Various event formats
+  ' TODO: Combine statements into one if statement.  Or something else like SWITCH?
+
+  if eventType = types.SKIPCARDSHOWN
+  end if
+
+  if eventType = types.ADFREEPOD
+    m.skipAds = true
+  end if
+
+  if eventType = types.ADCOMPLETED
+    restorePlayback()
+  end if
+
+  if eventType = types.USERCANCEL OR eventType = types.OPTOUT OR eventType = types.NOADSAVAILABLE OR eventType = types.ADERROR
+    cleanUpAdRenderer()
+    restorePlayback()
+  end if
+
+  if eventType = types.USERCANCELSTREAM
+    cleanUpAdRenderer()
+    m.top.playbackEvent = { trigger: "cancelStream" } ' Might want to change this API since this is the flows API to the scene
+  end if
+end sub
+
+sub handleAds(ads)
+  if ads <> invalid AND ads.ads.count() > 0
+    firstAd = ads.ads[0] 'Assume truex can only be first ad
+
+    if m.skipAds then
+      ads.viewed = true
+      m.skipAds = false
+
+      ' Show choice card if needed?
+      return
+    end if
```

Comment:
looks likes this skipAds logic is in the wrong place. (resume ads, perhaps?)

Reply:
This is intended, I've changed the logic for this a bit.  PR is a bit stale.